apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/UrlSigner.PostPolicy.cs

```diff
@@ -0,0 +1,315 @@
+            public void SetStartsWith(PostPolicyKnownElement element, string prefix)
+            {
+                GaxPreconditions.CheckEnumValue(element, nameof(element));
+                ISupportsStartsWith supportsStartsWith = s_knownElements[(int)element] as ISupportsStartsWith ??
+                    throw new ArgumentException($"Element {element} does not support the start-with condition.");
+
+                AddOrReplace(supportsStartsWith, new StartsWith<ISupportsStartsWith>(supportsStartsWith, prefix));
+            }
+
+            /// <summary>
+            /// Sets a range condition for <paramref name="element"/>.
+            /// If any condition existed on this instance for <paramref name="element"/> it will
+            /// be overriden with this range condition.
+            /// </summary>
+            /// <typeparam name="TValue">The type of the values defining the range.</typeparam>
+            /// <param name="element">The element for which to set a range condition.</param>
+            /// <param name="min">The range lower bound.</param>
+            /// <param name="max">The range upper bound.</param>
+            public void SetRange<TValue>(PostPolicyKnownElement element, TValue min, TValue max)
+                where TValue : IComparable
+            {
+                GaxPreconditions.CheckEnumValue(element, nameof(element));
+                ISupportsRange<TValue> supportsRange = s_knownElements[(int)element] as ISupportsRange<TValue> ??
+                    throw new ArgumentException($"Element {element} does not support range conditions for values of {typeof(TValue).FullName}.");
+
+                AddOrReplace(supportsRange, new InRange<ISupportsRange<TValue>, TValue>(supportsRange, min, max));
+            }
+
+            /// <summary>
+            /// Sets a required field value for <paramref name="element"/> of name <paramref name="elementName"/>.
+            /// If any condition existed on this instance for <paramref name="element"/> of name <paramref name="elementName"/>
+            /// it will be overriden with this required field value.
+            /// </summary>
+            /// <typeparam name="TValue">The type of the value to set the element to.</typeparam>
+            /// <param name="element">The element for which to set a required field value.</param>
+            /// <param name="elementName">The custom element name.</param>
+            /// <param name="value">The value to set the element to.</param>
+            /// <param name="valueToObject">The function to convert <paramref name="value"/> to the
+            /// representation which with it should be included in the policy. May be null in which case
+            /// <typeparamref name="TValue"/> ToString() will be used.</param>
+            public void SetCustomField<TValue>(PostPolicyCustomElement element, string elementName, TValue value, Func<TValue, object> valueToObject = null)
```

Comment:
Given that there are only two options here (Google metadata or genuinely custom) is the enum worth having rather than just two methods? Or is that part of the language-agnostic design requirements?